gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/HadoopFsEndPoint.java

```diff
@@ -51,12 +51,16 @@
 
   public abstract Config getSelectionConfig();
 
-  @Override
-  public boolean isAvailable() {
+  /**
+   * A helper utility for data/filesystem availability checking
+   * @param path The path to be checked. For fs availability checking, just use "/"
+   * @return If the filesystem/path exists or not.
+   */
+  public boolean availabilityChecker(Path path){
```

Comment:
for boolean methods, use an `is...` name, e.g. `isPathAvailable`, similar for `datasetAvailable`.